Libraries/src/Amazon.Lambda.Annotations.SourceGenerators/LambdaFunctionCodeGenerator.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System.Linq;
+using System.Text;
+using System.Security.Cryptography;
+
+namespace Amazon.Lambda.Annotations.SourceGenerators
+{
+    public class LambdaFunctionCodeGenerator
+    {
+        private readonly MethodDeclarationSyntax _lambdaFunctionMethodSyntax;
+        private readonly ClassDeclarationSyntax _startupSyntax;
+        private readonly GeneratorExecutionContext _context;
+        private IMethodSymbol _lambdaFunctionMethodModel;
+        private IMethodSymbol _configureMethodSymbol;
+
+        /// <summary>
+        /// Returns class name of the generated method.
+        /// </summary>
+        public string ClassName => $"{LambdaFunctionMethodSymbol.ContainingType.Name}_{LambdaFunctionMethodSymbol.Name}_Generated";
+
+        /// <summary>
+        /// Represents simplified Lambda function,
+        /// It may or may not have the parameters such as customer input and ILambdaContext.
+        /// </summary>
+        public IMethodSymbol LambdaFunctionMethodSymbol
+        {
+            get
+            {
+                if (_lambdaFunctionMethodModel == null)
+                {
+                    var methodModel = _context.Compilation.GetSemanticModel(_lambdaFunctionMethodSyntax.SyntaxTree);
+                    _lambdaFunctionMethodModel = methodModel.GetDeclaredSymbol(_lambdaFunctionMethodSyntax);
+                }
+
+                return _lambdaFunctionMethodModel;
+            }
+        }
+
+        public INamedTypeSymbol ReturnType => LambdaFunctionMethodSymbol.ReturnType as INamedTypeSymbol;
+
+        /// <summary>
+        /// Represents the Configure(IServiceCollection) method in the LambdaStartup attributed class.
+        /// </summary>
+        /// <exception cref="NotSupportedException">Thrown when multiple LambdaStartup attributed classes exist in the project.</exception>
+        public IMethodSymbol ConfigureMethodSymbol
+        {
+            get
+            {
+                if (_startupSyntax == null)
+                {
+                    return null;
+                }
+
+                if (_configureMethodSymbol == null)
+                {
+                    var iServiceCollectionSymbol =
+                        _context.Compilation.GetTypeByMetadataName(
+                            "Microsoft.Extensions.DependencyInjection.IServiceCollection");
+
+                    var classModel = _context.Compilation.GetSemanticModel(_startupSyntax.SyntaxTree);
+                    foreach (var member in _startupSyntax.Members.Where(member => member.Kind() == SyntaxKind.MethodDeclaration))
+                    {
+                        var methodSyntax = (MethodDeclarationSyntax)member;
+                        var methodSymbol = classModel.GetDeclaredSymbol(methodSyntax);
+                        if (methodSymbol != null
```

Comment:
I think this needs to check if the method is `public`. As I understand this is looking for any method that takes in a single parameter of type `IServiceCollection`. Is this better then looking for a specific named method like `ConfigiureServices?

Reply:
I initially thought of that but then realized, if the method won't be public, customers will get static analyzer warning which they are already familiar with.

Reply:
Will match the exact signature including the method name because customers can have multiple methods with same IServiceCollection parameter and we will end up using a wrong method in cases.